api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -221,6 +265,23 @@ type AzureManagedControlPlaneSpec struct {
 	// DisableLocalAccounts disables getting static credentials for this cluster when set. Expected to only be used for AAD clusters.
 	// +optional
 	DisableLocalAccounts *bool `json:"disableLocalAccounts,omitempty"`
+
+	// AutoUpgradeProfile - Profile of auto upgrade configuration.
+	// +optional
+	AutoUpgradeProfile *ManagedClusterAutoUpgradeProfile `json:"autoUpgradeProfile,omitempty"`
+}
+
+// ManagedClusterAutoUpgradeProfile - Auto upgrade profile for a managed cluster.
+type ManagedClusterAutoUpgradeProfile struct {
+	// NodeOSUpgradeChannel is a manner in which the OS on your nodes is updated. The default is NodeImage. Possible values include: NodeImage,Unmanaged,None
+	// +kubebuilder:validation:Enum=NodeImage;Unmanaged;None
+	// +optional
+	NodeOSUpgradeChannel *NodeOSUpgradeChannel `json:"nodeOSUpgradeChannel,omitempty"`
+
+	// UpgradeChannel - upgrade channel for auto upgrade. Possible values include: "node-image","none","patch","rapid","stable"
```

Comment:
```suggestion
	// UpgradeChannel - upgrade channel for auto upgrade. Possible values include: 'node-image','none','patch','rapid','stable'
```
Use single quotes for consistency with the rest of the repo

Reply:
ack